File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cs
public class Solution {
    public string RemoveDuplicates(string s) {
        Stack<char> stack = new Stack<char>();

        foreach(char c in s)
        {
            if(stack.TryPeek(out char d) && d == c)
            {
                stack.Pop();
            } else {
                stack.Push(c);
            }
        }
        
        return string.Concat(stack.Reverse());
    }
}

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.ts
function removeDuplicates(s: string): string {
    const stack: string[] = [];
    //Array.constructor.prototype.peek: any = () => {stack[stack.length-1] || ""};
    
    for(const c of s)
    {
        //stack.constructor.prototype.peek = () => stack[stack.length-1] || "";
        const peek = () => stack[stack.length-1] || "";
        
        if(peek() == c){
            stack.pop();
        } else {
            stack.push(c);
        }
    }
    return stack.join("");
};